index.bs

```diff
@@ -3798,168 +3991,788 @@ To <dfn>get the response data</dfn> given |response|:
+   <dt>Return Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      network.AddInterceptResult = {
+        intercept: network.Intercept
+      }
+    </pre>
+   </dd>
 </dl>
 
-This event is emitted before a request is sent (either over the network or
-before it's handled by a serviceworker or a local cache).
-
-<div algorithm>
-The [=remote end event trigger=] is the <dfn export>WebDriver BiDi before
-request sent</dfn> steps given |request|:
-
-1. If [=before request sent map=] does not contain |request|, set [=before
-   request sent map=][|request|] to a new set.
-
-1. Let |redirect count| be |request|'s [=redirect count=].
-
-1. Add |redirect count| to [=before request sent map=][|request|].
-
-1. Let (|related browsing contexts|, |params|) be the result of [=get the base
-   network event data=] with |request|, and |redirect count|.
+<div algorithm="remote end steps for network.addRequestIntercept">
+The [=remote end steps=] given |session| and |command parameters| are:
 
-1. Let |initiator| be the result of [=get the initiator=] with |request|.
+1. Let |intercept| be the string representation of a [[!RFC4122|UUID]].
 
-1. Set the <code>initiator</code> field of |params| to |initiator|.
+1. Let |url patterns| be the <code>urlPatterns</code> field of |command
+   parameters| if present, or null otherwise.
 
-1. Assert: |params| matches the <code>network.BeforeRequestSentParameters</code>
-   production.
+1. Let |intercept map| be |session|'s [=intercept map=].
 
-1. Let |body| be a map matching the <code>network.BeforeRequestSent</code>
-   production, with the <code>params</code> field set to |params|.
+1. Set |intercept map|[|intercept|] to a struct with <code>url patterns</code>
+   |url patterns| and <code>phases</code> |command parameters|["<code>phases</code>"].
 
-1. [=Emit events=] with |body| and |related browsing contexts|.
+1. Return a new [=/map=] matching the
+   <code>network.AddNetworkInterceptResult</code> with the
+   <code>intercept</code> field set to |intercept|.
 
 </div>
 
-#### The network.fetchError Event #### {#event-network-fetchError}
+#### The network.continueRequest Command ####  {#command-network-continueRequest}
+
+The <dfn export for=commands>network.continueRequest</dfn> command continues a request
+that's blocked by a [=network intercept=].
 
 <dl>
-   <dt>Event Type</dt>
+   <dt>Command Type</dt>
    <dd>
-      <pre class="cddl local-cddl">
-        network.FetchError = {
-         method: "network.fetchError",
-         params: network.FetchErrorParameters
-       }
+      <pre class="cddl remote-cddl">
+      network.ContinueRequest = {
+        method: "network.continueRequest",
+        params: network.ContinueRequestParameters
+      }
 
-       network.FetchErrorParameters = {
-         network.BaseParameters,
-         errorText: text,
-       }
+      network.ContinueRequestParameters = {
+        request: network.Request,
+        ?body: network.Body,
+        ?headers: [*network.Header],
```

Comment:
The `params` property of the `network.beforeRequestSent` event includes the properties of a `network.BaseParameters` object, one of which is the `request` property, the value of which is a `network.RequestData` object. The `network.RequestData` object has a `cookies` property which separates out the cookies of the request into a separate property. We don't allow separate setting of the cookies here. This seems asymmetrical, making it challenging to know how to manipulate add/edit/remove cookies from the request. It is possible to modify the request's cookies by creating a "Cookie" header, but that seems error prone, especially when needing to properly serialize the cookie value in the header.

Reply:
I've added an initial cut at handling cookies. Let me know what you think.

Reply:
I'm pretty happy with this approach. I'd guess that cookie manipulation is going to be the one thing people will want to do when monitoring/modifying network traffic more than just about anything else. This makes that easier.